### PR TITLE
Remove -ti options (from Makefile components that are on the critical path for image builds )

### PR DESCRIPTION
**Pull request by jayunit100**
Hi ! This is Part of  https://github.com/projectcalico/calico/issues/2936, where I'm building some automation tooling around calico.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -266,7 +266,7 @@ bin/dikastes-ppc64le: ARCH=ppc64le
 bin/dikastes-s390x: ARCH=s390x
 bin/dikastes-%: local_build vendor proto $(SRC_FILES)
 	mkdir -p bin
-	$(DOCKER_RUN_RO) -ti \
+	$(DOCKER_RUN_RO) \
```
Comment (fasaxc) [inline Makefile]: Doesn't removing `-ti` break `Ctrl-C`?

Comment (jayunit100) [reply Makefile]: - am assuming this is more of a backend build, so requiring a TTY is probably not really necessary ? i.e. its really just an image builder, presumeably, that isn't going to be interrupted in realtime, and isnt receiving any other input from the console... i.e. like  the builds in k8s (for example  https://github.com/kubernetes/kubernetes/blob/master/build/pause/Makefile ).

- I did a quick couple of quick builds of `app-policy` and it seems to generally run in < 30 seconds, so doubt any major interactive stuff is required?

- Also, many other `docker run` directives in the codebapth for `dev-image` don't use `-it`... so , kinda thinking  consistency  might be generally useful in addition to solving the selfish problem that im experiencing :).

Comment (rafaelvanoni) [reply Makefile]: Yeah, I agree it's not great to break `ctrl-c` but it doesn't work for the reasons @jayunit100 mentioned. I'm good with this fix.

Comment (jayunit100) [reply Makefile]: ok, assuming resolved.... thanks everyone.  ... im thinking uniforming the way we call docker across the board would be good for the future. maybe ill look into that at some point once i understand more about the overall build structure.

Comment (fasaxc) [reply Makefile]: OK

**Pull request merged**